cpp/src/ice2slice/Gen.cpp

```diff
@@ -12,26 +12,57 @@ using namespace IceInternal;
 
 namespace
 {
-    static string getCSharpNamespace(const ContainedPtr& cont, bool& hasCSharpNamespaceAttribute)
+    static void writeCsIdentifier(const ContainedPtr& cont, Output& out)
+    {
+        // Modules are not allowed to have a cs::identifier attribute in IceRPC Slice.
+        // they are handled separately in the getOutput function.
+        assert(!dynamic_pointer_cast<Module>(cont));
+
+        if (auto metadata = cont->getMetadataArgs("cs:identifier"))
+        {
+            out << nl << "[cs::identifier(\"" << *metadata << "\")]";
+        }
+    }
+
+    // We cannot use mappedName because it depends on the language name used to create the Unit.
+    static string getCsMappedName(const ContainedPtr& cont)
+    {
+        if (auto metadata = cont->getMetadataArgs("cs:identifier"))
+        {
+            return *metadata;
+        }
+        else
+        {
+            return cont->name();
+        }
+    }
+
+    static string getCsMappedScoped(const ContainedPtr& cont, const string& separator);
```

Comment:
`separator` will always be `"."` in the context that these functions are being called.
I'd remove this extra parameter and just hardcode `"."` in `getCsMappedScope`.